docs/user/syncronization.md

```diff
@@ -0,0 +1,142 @@
+The next group of manifests are under `0000_05_config-operator_*`.
+Because the number is bumped, the graph blocks until the previous `0000_03_*` are all complete before beginning the `0000_05_*` block.
+
+We are more relaxed for the initial install, because there is not yet any user data in the cluster to be worried about.
+So the graph nodes are all parallized with the by-number ordering flattened out:
+
+<div style="text-align:center">
+  <img src="tasks-flatten-by-number-and-component.svg" width="100%" />
+</div>
+
+For the usual reconciliation loop (neither an upgrade between releases nor a fresh install), the flattened graph is also randomly permuted to avoid hanging on ordering bugs.
+
+## Synchronizing the graph
+
+The cluster-version operator spawns worker goroutines that walk the graph, pushing manifests in their queue.
+For each manifest in the node, the worker synchronizes the cluster with the manifest using a resource builder.
+On error (or timeout), the worker abandons the manifest, graph node, and any dependencies of that graph node.
+On success, the worker proceeds to the next manifest in the graph node.
+
+## Resource builders
+
+Resource builders synchronize the cluster with a manifest from the release image.
+The general approach is to generates a merged manifest combining critical spec properties from the release-image manifest with data from a preexisting in-cluster object, if any.
+If the merged manifest differs from the in-cluster object, the merged manifest is pushed back into the cluster.
+
+Some types have additional logic, as described in the following subsections.
+Note that this logic only applies to manifests included in the release image itself.
+For example, only ClusterOperator from the release image will have the blocking logic described [below](#clusteroperator); if an admin or secondary operator pushed a ClusterOperator object, it would not impact the cluster-version operator's graph synchronization.
+
+### ClusterOperator
+
+After pushing the merged ClusterOperator into the cluster, the builder monitors the in-cluster object and blocks until it is:
+
+* Available
+* Either not progressing or listing at least one version.
+    The progressing check is deprecated and will be removed once all operators are reporting versions.
+* Not degraded (except during initiazation, where we ignore the degraded status)
+
+### CustomResourceDefinition
+
+After pushing the merged CustomResourceDefinition into the cluster, the builder monitors the in-cluster object and blocks until it is established.
+
+### DaemonSet
+
+After pushing the merged DaemonSet into the cluster, the builder monitors
+
+* FIXME: `UpdatedNumberScheduled == DesiredNumberScheduled` check?
+* Has no unavailable nodes.
+
+### Deployment
+
+After pushing the merged Deployment into the cluster (FIXME: `Generation > 1` check from 14fab0b2?), the builder monitors the in-cluster object and blocks until it:
```

Comment:
This comparison is happening [here][1] since 14fab0b2 (#26).  @abhinavdahiya, do you remember why you added it?  Is the idea to not wait on Deployments during the initial install phase?

[1]: https://github.com/openshift/cluster-version-operator/blame/5f087d4fea766fd563e22f2c07868de87e69ae1a/lib/resourcebuilder/apps.go#L136

Reply:
> Is the idea to not wait on Deployments during the initial install phase?

Yup.